holoviews/operation/datashader.py

```diff
@@ -0,0 +1,208 @@
+import param
+import numpy as np
+import pandas as pd
+import xarray as xr
+import datashader as ds
+import datashader.transfer_functions as tf
+
+import datashader.core
+from datashader.core import bypixel
+from datashader.pandas import pandas_pipeline
+from datashape.dispatch import dispatch
+from datashape import discover as dsdiscover
+
+from ..core import (ElementOperation, Element, Dimension, NdOverlay,
+                    Overlay, CompositeOverlay, Dataset)
+from ..core.util import get_param_values
+from ..element import GridImage, Path, Curve, Contours, RGB
+
+
+@dispatch(Element)
+def discover(dataset):
+    """
+    Allows datashader to correctly discover the dtypes of the data
+    in a holoviews Element.
+    """
+    return dsdiscover(dataset.dframe().head())
+
+
+@bypixel.pipeline.register(Element)
+def dataset_pipeline(dataset, schema, canvas, glyph, summary):
+    """
+    Defines how to apply a datashader pipeline to a holoviews Element,
+    using multidispatch. Returns an Image type with the appropriate
+    bounds and dimensions. Passing the returned Image to datashader
+    transfer functions is not yet supported.
+    """
+    x0, x1 = canvas.x_range
+    y0, y1 = canvas.y_range
+    kdims = [dataset.get_dimension(d) for d in (glyph.x, glyph.y)]
+
+    column = summary.column
+    if column and isinstance(summary, ds.count_cat):
+        name = '%s Count' % summary.column
+    else:
+        name = column
+    vdims = [dataset.get_dimension(column)(name) if column
+             else Dimension('Count')]
+
+    aggregate = pandas_pipeline(dataset.dframe(), schema, canvas,
+                                glyph, summary)
+    aggregate = aggregate.rename({'x_axis': kdims[0].name,
+                                  'y_axis': kdims[1].name})
+
+    params = dict(get_param_values(dataset), kdims=kdims,
+                  datatype=['xarray'], vdims=vdims)
+
+    if aggregate.ndim == 2:
+        return GridImage(aggregate, **params)
+    else:
+        return NdOverlay({c: GridImage(aggregate.sel(**{column: c}),
+                                       **params)
+                          for c in aggregate.coords[column].data},
+                         kdims=[dataset.get_dimension(column)])
+
+
+class Aggregate(ElementOperation):
+
+    aggregator = param.ClassSelector(class_=ds.reductions.Reduction,
+                                     default=ds.count())
+
+    height = param.Integer(default=800)
+
+    width = param.Integer(default=600)
+
+    x_range  = param.NumericTuple(default=None, length=2,
+                                  allow_None=True)
+
+    y_range  = param.NumericTuple(default=None, length=2,
+                                  allow_None=True)
+
+    x_sampling = param.Number(default=None, doc="""
+        Specifies the smallest allowed sampling interval along the y-axis.""")
+
+    y_sampling = param.Number(default=None, doc="""
+        Specifies the smallest allowed sampling interval along the y-axis.""")
+
```

Comment:
They let you define a minimum sampling interval. It's one of my major annoyances with datashader you can zoom in beyond the sampling resolution, giving you useless speckly aggregates.